http-clients/src/test/java/com/palantir/remoting/http/FeignClientFactoryTest.java

```diff
@@ -59,20 +69,44 @@ public void testSslSocketFactory_cannotConnectWhenSocketFactoryIsNotSet() throws
 
     @Test
     public void testSslSocketFactory_canConnectWhenSocketFactoryIsSet() throws Exception {
+        TestEchoService service = createProxy();
+        assertThat(service.echo("foo"), is("foo"));
+    }
+
+    @Test
+    public void testBraveTracing() throws Exception {
+        TestEchoService service = createProxy();
+        service.echo("foo"); // TODO(rfink) This prints stuff to the log. Verify it actually does so.
+    }
+
+    private TestEchoService createProxy() {
         String endpointUri = "https://localhost:" + APP.getLocalPort();
         SslConfiguration sslConfig = SslConfiguration.of(Paths.get("src/test/resources/trustStore.jks"));
-        TestEchoService service = FeignClients.standard("test suite user agent")
+        return FeignClients.standard("test suite user agent")
                 .createProxy(
                         Optional.of(SslSocketFactories.createSslSocketFactory(sslConfig)),
                         endpointUri,
                         TestEchoService.class);
-        assertThat(service.echo("foo"), is("foo"));
     }
 
+
     public static final class TestEchoServer extends Application<Configuration> {
         @Override
         public void run(Configuration config, final Environment env) throws Exception {
             env.jersey().register(new TestEchoResource());
+            ServerTracer serverTracer = ServerTracer.builder()
+                    .traceSampler(Sampler.ALWAYS_SAMPLE)
+                    .randomGenerator(new Random())
+                    .state(new ThreadLocalServerClientAndLocalSpanState(0, 1, "server"))
```

Comment:
IP address and port. I just filled in dummy values for now, these are not
here to stay.

On Tue, Jun 21, 2016 at 7:47 PM Mark Elliot <notifications@github.com>
wrote:

> In
> http-clients/src/test/java/com/palantir/remoting/http/FeignClientFactoryTest.java
> <https://github.com/palantir/http-remoting/pull/139#discussion_r67987166>:
>
> >      public static final class TestEchoServer extends Application<Configuration> {
> >          @Override
> >          public void run(Configuration config, final Environment env) throws Exception {
> >              env.jersey().register(new TestEchoResource());
> > +            ServerTracer serverTracer = ServerTracer.builder()
> > +                    .traceSampler(Sampler.ALWAYS_SAMPLE)
> > +                    .randomGenerator(new Random())
> > +                    .state(new ThreadLocalServerClientAndLocalSpanState(0, 1, "server"))
>
> kind of curious what these magic numbers are
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/139/files/234359b4fb4c9bcb2e2b1ae05d2e0e432fecc73a#r67987166>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGOdwaWiT7RoU5r_bsSr8MXfl0NSG1qYks5qOKIdgaJpZM4I7VY4>
> .
>